apps/web-client/src/lib/db/cr-sqlite/note.ts

```diff
@@ -50,6 +51,231 @@ export async function getAllInboundNotes(db: DB): Promise<InboundNoteListItem[]>
 	return res.map(({ updated_at, ...el }) => ({ ...el, updatedAt: new Date(updated_at), totalBooks: 0 }));
 }
 
-export function deleteNote(db: DB, id: number) {
+type GetNoteResponse = {
+	id: number;
+	displayName: string;
+	/** for inbound notes, undefined otherwise */
+	warehouseId?: number;
+	/** for inbound notes, undefined otherwise */
+	warehouseName?: string;
+	/** for outbound notes, undefined otherwise */
+	defaultWarehouse?: number;
+	// if note has warehouseId, this should be: "inbound"
+	// if note is a reconciliation note, this should be: "inbound"
+	// in all other cases, this should be: "outbound"
+	noteType: string;
+	updatedAt: Date;
+	committed: boolean;
+	committedAt?: Date;
+};
+
+export async function getNoteById(db: DB, id: number): Promise<GetNoteResponse | undefined> {
+	const query = `
+		SELECT
+			note.id,
+			note.display_name AS displayName,
+			note.warehouse_id AS warehouseId,
+			warehouse.display_name AS warehouseName,
+			note.updated_at,
+			note.committed,
+			note.committed_at,
+			note.is_reconciliation_note
+		FROM note
+		LEFT JOIN warehouse ON note.warehouse_id = warehouse.id
+		WHERE note.id = ?
+	`;
+
+	const result = await db.execO<{
+		id: number;
+		displayName: string;
+		warehouseId?: number;
+		warehouseName?: string;
+		is_reconciliation_note: number;
+		updated_at: number;
+		committed: number;
+		committed_at: number;
+	}>(query, [id]);
+
+	if (result.length === 0) {
+		return undefined;
+	}
+
+	const { updated_at, committed_at, committed, is_reconciliation_note, ...note } = result[0];
+
+	const noteType = note.warehouseId || is_reconciliation_note ? "inbound" : "outbound";
+
+	return {
+		...note,
+		noteType,
+		updatedAt: new Date(updated_at),
+		committed: Boolean(committed),
+		committedAt: committed_at ? new Date(committed_at) : undefined
+	};
+}
+
+// NOTE: default warehouse is not used at the moment (it will be used for outbound notes)
+export async function updateNote(db: DB, id: number, payload: { displayName?: string; defaultWarehouse?: number }): Promise<void> {
+	const note = await getNoteById(db, id);
+	if (note?.committed) {
+		console.warn("Trying to update a committed note: this is a noop, but probably indicates a bug in the calling code.");
+		return;
+	}
+
+	const { displayName, defaultWarehouse } = payload;
+
+	const updateFields = [];
+	const updateValues: (string | number)[] = [];
+
+	if (displayName !== undefined) {
+		updateFields.push("display_name = ?");
+		updateValues.push(displayName);
+	}
+
+	if (defaultWarehouse !== undefined) {
+		updateFields.push("default_warehouse = ?");
+		updateValues.push(defaultWarehouse);
+	}
+
+	if (updateFields.length === 0) {
+		return;
+	}
+
+	updateFields.push("updated_at = (strftime('%s', 'now') * 1000)");
```

Comment:
Since we're not in a trigger (and we found out we can't really use those for `updatedAt`) we should probably use a higher resolution value provided from the Javascript clock.

Makes sense as a separate task. I'll update #647